keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -1768,6 +1769,57 @@ of 1 bit (in the terms of the original networking setting).  That is
 isomorphic to (unweighted) fair queuing!  The weighted versions, still
 with miniscule quanta, are also isomorphic.
 
+#### Avoiding Disastrous Lockout
+
+We want to allow admins to edit the configuration, but we do not want
+these edits to be able to cause critical problems.  The critical
+problems are locking out administrators.  That would include locking
+out secondary requests in service of such primary requests.  By
+"locking out" we mean a total stop or an extreme slowdown.  We favor
+the approach of identifying a category of critical requests and
+ensuring that they are not held up by this feature.  This approach
+requires answering two questions: how do we identify the requests that
+must be let through, and what feature(s) of the configuration and
+behavior cause such requests to always get through?
+
+The likely way to identify the requests to always let through is by
+user group.  The group `system:masters` is a good place to start.  It
```

Comment:
system:masters for human admins, and apiserver self-requests (@deads2k probably knows the right way to identify these) for secondary requests implied by human requests. I think that should be sufficient.

Reply:
I am not sure what you mean by "secondary requests implied by human requests".  The scenario that concerns me is when an admission webhook or any other call-out is configured into the request handling chain, and a human makes a primary request, and that call-out is done, and the server handling that call turns around and makes a synchronous call back (i.e., a secondary request) to some apiserver --- we currently have no guarantee that this secondary request will authenticate with group `system:masters`.

Reply:
> when an admission webhook or any other call-out is configured into the request handling chain

in fact, most webhooks/aa-servers are deployed in-cluster hence accessing apiserver via service-accounts. `system:masters` won't cover them i suppose

Reply:
I don't recall agreeing to that.  I am not entirely sure what you mean.  Do you mean modifying everything in the handler chain that makes a call-out so that it does _not_ make the call-out if the request's target object is an API Priority and Fairness API object?

Reply:
I mean doing whatever it is we do to prevent people from webhooking webhookconfigurations and breaking the cluster that way.